### PR TITLE
N3LO singlet anomalous dimensions nf^3

### PR DESCRIPTION
**Pull request by giacomomagni**
Implementation of N3LO Singlet anomalous dimensions  part proportional to `nf^3` which is known exact.

### CODE REVIEW

File: doc/source/theory/N3LO_ad.rst [hunk 1]
```diff
@@ -142,3 +142,53 @@ In |EKO| they are implemented as follows:
...
+            - |T|
+            - |T|
+
+        *   - :math:`\gamma_{qg}^{(3)}`
+            - |T|
+            - |T|
+            - |T|
+            - |T|
+
+Only the parts proportional to :math:`n_f^3` are known analytical
```
Comment (felixhekhorn) [inline doc/source/theory/N3LO_ad.rst:192]:
```suggestion
Only the parts proportional to :math:`n_f^3` are known analytically
```

**Review by felixhekhorn — APPROVED** [review]
Again is mostly mathematics so should be alright ...

File: src/eko/anomalous_dimensions/as4/__init__.py [hunk 2]
```diff
@@ -4,6 +4,55 @@
...
+            |N3LO| singlet anomalous dimension matrix
+            :math:`\gamma_{S}^{(3)}(N)`
+
+      See Also
+      --------
+        gamma_nsp : :math:`\gamma_{ns,+}^{(3)}`
+        gamma_ps : :math:`\gamma_{ps}^{(3)}`
+        gamma_qg : :math:`\gamma_{qg}^{(3)}`
+        gamma_gq : :math:`\gamma_{gq}^{(3)}`
+        gamma_gg : :math:`\gamma_{gg}^{(3)}`
```
Comment (AleCandido) [inline src/eko/anomalous_dimensions/as4/__init__.py:51]: Just a small remark: this docstring is not fully compliant with Numpy format.

File: tests/eko/test_ad_as4.py [hunk 3]
```diff
@@ -39,6 +48,36 @@ def test_quark_number_conservation():
...
+    N = 2
+    sx_cache = compute_harmonics_cache(N, 5, True)
+
+    # nf^3 part
+    np.testing.assert_allclose(
+        gnsp.gamma_ns_nf3(N, sx_cache)
+        + gps.gamma_ps_nf3(N, sx_cache)
+        + ggq.gamma_gq_nf3(N, sx_cache),
+        0,
+        atol=3e-15,
```
Comment (AleCandido) [inline tests/eko/test_ad_as4.py:61]: This is wonderful :D

**Pull request merged**